esp-hal-rom/src/lib.rs

```diff
@@ -0,0 +1,57 @@
+#![no_std]
+
+#[doc(hidden)]
+/// Helper macro for checking doctest code snippets
+#[macro_export]
+macro_rules! before_snippet {
+    () => {
+        r#"
+# #![no_std]
+# use procmacros::handler;
+# use esp_hal::{interrupt::{self, InterruptConfigurable}, time::{Duration, Instant, Rate}};
+# macro_rules! println {
+#     ($($tt:tt)*) => { };
+# }
+# macro_rules! print {
+#     ($($tt:tt)*) => { };
+# }
+# #[panic_handler]
+# fn panic(_ : &core::panic::PanicInfo) -> ! {
+#     loop {}
+# }
+# fn main() {
+#   let _ = example();
+# }
+# struct ExampleError {}
+# impl <T> From<T> for ExampleError where T: core::fmt::Debug {
+#   fn from(_value: T) -> Self {
+#       Self{}
+#   }
+# }
+# async fn example() -> Result<(), ExampleError> {
+#   let mut peripherals = esp_hal::init(esp_hal::Config::default());
+"#
+    };
+}
+
+pub mod rom;
+
+/// This is needed by `libesp_rom.a` (if used)
+/// Other crates (i.e. esp-wifi) also rely on this being defined somewhere
+#[unsafe(no_mangle)]
+unsafe extern "C" fn __assert_func(
```

Comment:
Suppose the leading underscores are used as an indication to the function being private? Perhaps add `#[doc(hidden)]`?

Reply:
it's not public so not a problem